Identifying changes (#304)
bcgov/platform-developer-docs: src/docs/app-monitoring/managing-resource-quotas-in-kubernetes.md

### Definitions

- **CPU requests:** The guaranteed amount of CPU for a container in Kubernetes. It helps the scheduler place pods on nodes that have enough available CPU

> - **CPU limits:** The maximum amount of CPU a container can use. If the container exceeds this limit, it gets slowed down and can't use more than its allocated share. CPU limits are no longer required on BC Gov's OpenShift platform. For perfomance reasons, it is best to avoid setting CPU limits or remove existing CPU limits. 

Author: ```suggestion
- **CPU limits:** The maximum CPU a container can use. If the container exceeds this limit, it slows down and cannot exceed its allocated share. CPU limits are no longer required on BC Gov's OpenShift platform. To improve performance, avoid setting CPU limits or remove any existing ones.
```

Reviewer: I've made another suggestion below for this.

> - **CPU limits:** The maximum amount of CPU a container can use. If the container exceeds this limit, it gets slowed down and can't use more than its allocated share. CPU limits are no longer required on BC Gov's OpenShift platform. For perfomance reasons, it is best to avoid setting CPU limits or remove existing CPU limits. 

Reviewer: ```suggestion
- **CPU limits:** The maximum amount of CPU a container can use. If the container exceeds this limit, it is throttled  by OpenShift and cannot exceed its allocated share of CPU. CPU limits are no longer required on BC Gov's OpenShift platform. It is best to avoid setting CPU limits or remove existing CPU limits. Avoiding CPU limits can offer improved performance in some situations. 
```

spec:
  hard:
    requests.cpu: "4"   # Total CPU requests for all pods.
    requests.memory: "8Gi"  # Total memory requests.
```
The total sum of resources `requests.cpu` and `requests.memory` for containers in your namespace can not exceed the thresholds defined in the quota. 

### Case studies

**Scenario: Resolving CPU over-provisioning**

> A SaaS company noticed they weren't fully utilizing their allocated resources. They set up monitoring tools to study usage patterns and adjusted their ResourceQuota accordingly. This move significantly cut costs without affecting performance. In the BCGov context, resource quotas are adjusted via the [Product Registry](https://registry.developer.gov.bc.ca/) and then automatically applied to OpenShift. 

Author: ```suggestion
A SaaS company noticed they weren't fully utilizing their allocated resources. They set up monitoring tools to study usage patterns and adjusted their ResourceQuota accordingly. This move significantly cut costs without affecting performance. In the BCGov context, resource quotas are adjusted via the [Platform Product Registry](https://registry.developer.gov.bc.ca/) and then automatically applied to OpenShift. 
```

Reviewer: Not sure about this one- it's called the Product Registry at the top of the page, but then has 'Welcome to B.C. Gov's Platform Product Registry' further down.  It does now also manage products that are not running on BC Gov's platform eg. AWS.

Author: I believe we have called it in several documents and public cloud: Platform Product Registry

To manage CPU throttling effectively, it is essential to adjust both the application and Kubernetes settings:

> * **Remove CPU limits**: If you've previously set up `limit.cpu` for you containers, it is best to remove this for performance reasons. 

Author: ```suggestion
* **Remove CPU limits**: If you previously set `limit.cpu` for your containers, remove it to improve performance.
```
